hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OzoneContainer.java

```diff
@@ -484,6 +485,11 @@ public void start(String clusterId) throws IOException {
       return;
     }
 
+    DatanodeLayoutStorage layoutStorage
+        = new DatanodeLayoutStorage(config);
+    layoutStorage.setClusterId(clusterId);
+    layoutStorage.persistCurrentState();
```

Comment:
The VERSION file will be overwritten for each restart.  Unfortunately, it is the current behavior.

The correct behavior is that the VERSION file should 
- be created when storage is formatted, and
- be overwritten when there is a layout change.

For the other cases, the VERSION file should not be changed, i.e. no overwriting the file is needed.

Let change it separately.